docs/DeveloperGuide.md

```diff
@@ -1,22 +1,241 @@
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The sequence diagram below demonstrates the interactions within the `Parser` component when a user inputs the command: `add id/A1234567I gpa/5.0 p/{13,61,43}`.
 
+![ParserSequence](UML_Diagrams/ParserSequence.drawio.svg)
+
+The boolean return value of `parseInput()` indicates whether the user has chosen to continue or terminate the program. A `true` value keeps FindOurSEP running, while a `false` value ends the program.
+
+### Commands
+
+#### Add Command
+
+#### Delete Command
+
+#### Find Command
+
+#### Filter Command
+
+#### Criteria Command
+
+#### Stats Command
+
+#### ViewQuota Command
+
+#### List Command
+
+#### Allocate Command
+
+#### Revert Command
+
+![RevertSequence](/docs/UML_Diagrams/RevertCommandSequence.drawio.svg)
+
+Upon parsing a `revert` command, a `RevertCommand` instance is created. `RevertCommand` then calls the `revertAllocation()`
+method in `StudentList`, which loops through all the students in the `students` array list. The method `revertAllocation()`
+within the `Student` objects resets the allocation status and allocated university. The operation is completed by calling
+the `UI` to print the templated response from the `Messages` enum.
+
+#### Exit Command
+
+#### Help Command
+
+#### Generate Command
+
+![GenerateSequence](/docs/UML_Diagrams/GenerateCommandSequence.drawio.svg)
+
+The `generate` command is calls the `generateReport()` method in `StudentList`, which then calls the `generateReport()`
+in the `UI` using the student array list, which prints an ASCII table representing the allocation outcome.
+
+#### Unknown Command
+
+#### Exit Command
+
+### Allocator
+
+The `Allocator` class is responsible for allocating students to universities based on their preferences and GPA. It interacts with the `StudentList`, `UniversityRepository`, and `Student` classes to perform the allocation.
+
+In ``V1.0`` version, the allocation logic is designed as follows:
+
+* **Sorting by GPA**: The list of students is sorted in descending order of GPA, so higher-GPA students are prioritized.
+* **Preference-Based Allocation**: For each student:
+  * Iterates through the student’s university preferences.
+  * Checks if the university has open spots and if the student’s GPA meets the `minimumGPA` requirement.
+  * If both conditions are met, the student is allocated to that university, and the university’s spot count is reduced.
+  * The allocation stops once a student is assigned to a university.
+* **Sorting by ID**: After allocation, the list is re-sorted by student ID.
+
+Note that ``Allocator`` will copy the passed student list, therefore any modifications to the student list inside ``Allocator`` will not reflect in the original one.
+
+Here is a class diagram highlighting the fundamental structure of the `Allocator` class.
+
+![AllocatorClass](UML_Diagrams/AllocatorClass.drawio.svg)
```

Comment:
![image](https://github.com/user-attachments/assets/e66ee493-4481-4c0c-9665-0e77fcee1492)

The Solid Arrow for the class diagrams is incorrect. It should be a solid Triangle. You can refer to the Parser class diagram for the correct implementation.